FortnitePorting/Plugins/Blender/import_task.py

```diff
@@ -563,14 +591,153 @@ def get_meta(search_props):
         match mesh_type:
             case "Body":
                 meta = get_meta(["SkinColor"])
-            case "Head":
-                meta = get_meta(["MorphNames", "HatType", "PoseData"])
+            case "Head" | "Face":
+                meta = get_meta(["MorphNames", "HatType", "PoseData", "ReferencePose"])
+                is_face = mesh_type == "Face"
 
                 shape_keys = imported_mesh.data.shape_keys
                 if (morph_name := meta.get("MorphNames").get(meta.get("HatType"))) and shape_keys is not None:
                     for key in shape_keys.key_blocks:
                         if key.name.casefold() == morph_name.casefold():
                             key.value = 1.0
+
+                if (pose_data := meta.get("PoseData")):
+                    original_mode = bpy.context.active_object.mode
+                    try:
+                        bpy.ops.object.mode_set(mode='OBJECT')
+
+                        # Find the ARMATURE modifier
+                        armature_modifiers = [mod for mod in imported_mesh.modifiers if mod.type == "ARMATURE"]
+                        assert len(armature_modifiers) == 1, "expected at least one armature modifier"
+                        armature_modifier: bpy.types.ArmatureModifier = armature_modifiers[0]
+
+                        # Find the ARMATURE
+                        armatures = [
+                            ob
+                            for ob in bpy.data.objects
+                            if ob.type == "ARMATURE" and ob.name.split('.')[0] == imported_mesh.name.split('.')[0]
+                        ]
+                        assert len(armatures) == 1, "expected at least one armature"
+                        armature: bpy.types.Object = armatures[0]
+
+                        # Grab reference pose data
+                        reference_pose = meta.get("ReferencePose", dict())
+
+                        # Check how many bones are scaled in the reference pose
+                        face_attach_scale = Vector((1, 1, 1))
+                        for entry in reference_pose:
+                            scale = make_vector(entry['Scale'])
+                            if scale != Vector((1, 1, 1)):
+                                if entry['BoneName'] == 'faceAttach':
+                                    face_attach_scale = scale
+                                Log.warn(f"Non-zero scale: {entry}")
```

Comment:
Exporting Ranger, I notice the warning appearing for bones other than `faceAttach`:
```
[FNPORTING] Non-zero scale: {'BoneName': 'R_eye_lid_upper_mid', ... 'Scale': {'X': 0.999998, 'Y': 1.0000657, 'Z': 0.9999363}}
[FNPORTING] Non-zero scale: {'BoneName': 'R_eye_lid_lower_mid', ..., 'Scale': {'X': 0.999998, 'Y': 1.0000249, 'Z': 0.9999771}}
```

They're close enough to `(1, 1, 1)` that it _probably_ wont make a difference, however, I wonder if other characters have weird scaling in their reference pose which also needs to be accounted for when creating location data.

Just want to reiterate that the scaled location is based on observation from an old export of facial data I have, so this _may_ be inaccurate. I guess we can leave it in for now if there are no objections though?

Reply:
i mean if it works it works, until we encounter a breaking issue i think it's fine